src/script/run-make.sh

```diff
@@ -55,6 +55,8 @@ function prepare() {
 
         if in_jenkins; then
             if ! type clang-19 > /dev/null 2>&1 ; then
+                ci_debug "Removing existing llvm packages"
+                $SUDO apt-get purge --auto-remove llvm python3-lldb-13 llvm-13 -y
```

Comment:
Thanks for looking into this!
Should we check that clang-13 exists in the first place? After the first runs of this script it shouldn't exists anymore.
Alternatively, we could also clean our machines manually from clang-13 instead of adding this to the script IIUC.

Reply:
from https://tracker.ceph.com/issues/70792#note-2:
> The right solution is to fix the 'update clang' build script snippet to remove any other clang versions. This may happen again as we change the required version; removing 13 (or any other versions) now won't fix the problem, if there is one, when we move from 19 to 20, for example.

@dmick doesn't want to manage the builder dependencies manually. when we eventually want to bump this llvm version to 20+, we're likely to hit the same conflicts again

removing _any_ llvm packages is probably the most reliable approach here (ie purging `llvm python3-lldb` without specifying version), though it would uninstall/reinstall v19 packages for each build

Reply:
ubuntu 24.04 builders probably won't default to clang-13 anyway, so we may have conflicts with other versions too

Reply:
It seems to me that we want to purge any/all earlier versions of llvm-and-friends, not just 13

Reply:
> Do we know whether 14 would also conflict with 19?

unknown, but https://github.com/llvm/llvm-project/issues/64182 is about conflicts between 14 and 15/16

Reply:
> It seems to me that we want to purge any/all earlier versions of llvm-and-friends, not just 13

updated to remove the `-13`s from the apt purge command

Reply:
Do we think it's safer to leave the other llvm packages intact, or remove them as well?

Reply:
hmm looking at llvm.sh, this is the list of packages it installs:
> PKG="clang-$LLVM_VERSION lldb-$LLVM_VERSION lld-$LLVM_VERSION clangd-$LLVM_VERSION"

so i could copy those for the purge command. the conflicting python3-lldb isn't mentioned there, but i assume it would be removed along with 'lldb'?

Reply:
and, by the way, the llvm.sh on download.ceph.com is already out of date

Reply:
could include python3-lldb anyway to be safe, like this?
```
$DRY_RUN $SUDO apt-get purge --auto-remove clang lldb lld clangd python3-lldb -y
```